components/Common/BlogPostCard/index.tsx

```diff
@@ -0,0 +1,83 @@
+import { useMemo } from 'react';
+import type { ComponentProps, FC } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import AvatarGroup from '@/components/Common/AvatarGroup';
+import Preview from '@/components/Common/Preview';
+
+import styles from './index.module.css';
+
+const dateTimeFormat = new Intl.DateTimeFormat(navigator.language, {
+  day: 'numeric',
+  month: 'short',
+  year: 'numeric',
+});
+
+type Author = {
+  fullName: string;
+  src: ComponentProps<typeof AvatarGroup>['avatars'][number]['src'];
+};
+
+export type BlogPostCardProps = {
+  title: ComponentProps<typeof Preview>['title'];
+  type: Required<ComponentProps<typeof Preview>>['type'];
+  description: string;
+  authors: Author[];
+  date: Date;
+};
+
+const BlogPostCard: FC<BlogPostCardProps> = ({
+  title,
+  type,
+  description,
+  authors,
+  date,
+}) => {
+  const avatars = useMemo(() => {
+    const parsedAvatars = authors.map(({ fullName, src }) => ({
```

Comment:
Is a variable assignment really needed? Can't this be just inline without return statement and variable assignment?

```tsx
const avatars = useMemo(() => authors.map(...), [authors]);
```

Reply:
Thank you! The variable is indeed not needed. We can definitely inline `authors.map`.